integration-tests/tests/src/tests/counter.ts

```diff
@@ -0,0 +1,994 @@
+}
+
+function expectKeys(dictionary: Realm.Dictionary, keys: string[]) {
+  expect(Object.keys(dictionary)).members(keys);
+}
+
+describe("Counter", () => {
+  openRealmBeforeEach({
+    schema: [
+      WithCounterSchema,
+      WithNullableCounterSchema,
+      WithDefaultCounterSchema,
+      WithRegularIntSchema,
+      WithMixedSchema,
+    ],
+  });
+
+  const initialValuesList = [-100, 0, 1.0, 1000] as const;
+
+  describe("Create and access", () => {
+    describe("Via 'realm.create()'", () => {
+      it("can create and access (input: number)", function (this: RealmContext) {
+        for (let i = 0; i < initialValuesList.length; i++) {
+          const input = initialValuesList[i];
+          const { counter } = this.realm.write(() => {
+            return this.realm.create<IWithCounter>(WithCounterSchema.name, {
+              counter: input,
+            });
+          });
+
+          const expectedNumObjects = i + 1;
+          expect(this.realm.objects(WithCounterSchema.name).length).equals(expectedNumObjects);
+          expectCounter(counter);
+          expect(counter.value).equals(input);
+        }
+      });
+
+      it("can create and access (input: Counter)", function (this: RealmContext) {
+        const initialNumValues = initialValuesList;
+        const initialCounterValues: Counter[] = [];
+
+        // First create Realm objects with counters.
+        this.realm.write(() => {
+          for (const input of initialNumValues) {
+            const { counter } = this.realm.create<IWithCounter>(WithCounterSchema.name, {
+              counter: input,
+            });
+            expectCounter(counter);
+            expect(counter.value).equals(input);
+
+            initialCounterValues.push(counter);
+          }
+        });
+
+        // Use the managed Counters as input, each in a separate transaction.
+        for (let i = 0; i < initialCounterValues.length; i++) {
+          const input = initialCounterValues[i];
+          const { counter } = this.realm.write(() => {
+            return this.realm.create<IWithCounter>(WithCounterSchema.name, {
+              counter: input,
+            });
+          });
+
+          const expectedNumObjects = initialNumValues.length + i + 1;
+          expect(this.realm.objects(WithCounterSchema.name).length).equals(expectedNumObjects);
+          expectCounter(counter);
+          expect(counter.value).equals(input.value);
+        }
+      });
+
+      it("can create and access (input: default value)", function (this: RealmContext) {
+        const { counterWithDefault } = this.realm.write(() => {
+          // Pass an empty object in order to use the default value from the schema.
+          return this.realm.create<IWithDefaultCounter>(WithDefaultCounterSchema.name, {});
+        });
+
+        expect(this.realm.objects(WithDefaultCounterSchema.name).length).equals(1);
+        expectCounter(counterWithDefault);
+        expect(counterWithDefault.value).equals(0);
```

Comment:
As per my earlier comment, I would expect the default to be `0` even if not provided explicitly through the schema.